subsys/bluetooth/services/hids_c.c

```diff
@@ -213,6 +213,342 @@ static u16_t chrc_value_handle_by_uuid(struct bt_gatt_dm *dm,
+	if (ret) {
+		hids_prep_error(hids_c, ret);
+		return BT_GATT_ITER_STOP;
+	}
+
+	return BT_GATT_ITER_STOP;
+}
+
+/**
+ * @brief HIDS information read
+ *
+ * HIDS information read is started after connection to HID server.
+ *
+ * @param conn   Connection handler.
+ * @param err    Read ATT error code.
+ * @param params Notification parameters structure - the pointer
+ *               to the structure provided to read function.
+ * @param data   Pointer to the data buffer.
+ * @param length The size of the received data.
+ *
+ * @retval BT_GATT_ITER_STOP     Stop notification
+ * @retval BT_GATT_ITER_CONTINUE Continue notification
+ */
+static u8_t hid_info_read_process(struct bt_conn *conn, u8_t err,
+				   struct bt_gatt_read_params *params,
+				   const void *data, u16_t length);
+
+static int hid_info_read_start(struct bt_gatt_hids_c *hids_c)
+{
+	int err;
+
+	__ASSERT_NO_MSG(hids_c);
+	if (hids_c->handlers.info == 0) {
+		LOG_ERR("Device ready without HID information characteristic");
+		return -EINVAL;
+	}
+	LOG_DBG("HID information read start");
+	hids_c->read_params.func = hid_info_read_process;
+	hids_c->read_params.handle_count  = 1;
+	hids_c->read_params.single.handle = hids_c->handlers.info;
+	hids_c->read_params.single.offset = 0;
+	err = bt_gatt_read(hids_c->conn, &(hids_c->read_params));
+	if (err) {
+		LOG_ERR("HID information read error (err: %d)", err);
+		return err;
+	}
+	return 0;
+}
+
+static u8_t hid_info_read_process(struct bt_conn *conn, u8_t err,
+				   struct bt_gatt_read_params *params,
+				   const void *data, u16_t length)
+{
+	struct bt_gatt_hids_c *hids_c;
+	const u8_t *bdata = data;
+
+	hids_c = CONTAINER_OF(params,
+			      struct bt_gatt_hids_c,
+			      read_params);
+
+	if (err) {
+		LOG_ERR("HID information read error");
+		hids_prep_error(hids_c, err);
+		return BT_GATT_ITER_STOP;
+	}
+	if (length != 4 || !data) {
+		LOG_ERR("Unexpected HID information size: %u", length);
+		hids_prep_error(hids_c, -ENOTSUP);
+		return BT_GATT_ITER_STOP;
+	}
+
+	hids_c->info_val.bcd_hid = (u16_t)bdata[0] | (((u16_t)bdata[1]) << 8);
+	hids_c->info_val.b_country_code = bdata[2];
+	hids_c->info_val.flags = bdata[3];
+
+	LOG_DBG("HID information success:");
+	LOG_DBG("  bcdHID: %x", hids_c->info_val.bcd_hid);
+	LOG_DBG("  bCountryCode: 0x%x", hids_c->info_val.b_country_code);
+	LOG_DBG("  Flags: 0x%x", hids_c->info_val.flags);
+
+	err = repref_read_start(hids_c, 0);
+	if (err) {
+		hids_prep_error(hids_c, err);
+	}
+
+	return BT_GATT_ITER_STOP;
+}
+
+/**
+ * @brief Start anything that should be started after discovery
+ *
+ * This function is called when handles assign function finishes and
+ * starts the reading of the additional data from the HID server
+ * that is required for HIDS client to work.
+ *
+ * @param hids_c  See @ref bt_gatt_hids_c_handles_assign.
+ *
+ * @return 0 or negative error value.
+ */
+static int discovery_afterparty_start(struct bt_gatt_hids_c *hids_c)
```

Comment:
quite funny name, but I suggest to change it to ```post_discovery_operations_start``` or sth similar

Reply:
Changed to post_discovery_start